### PR TITLE
Module 1 Project Exercise

### PR DESCRIPTION
**Pull request by Jonesey13**
Creating this PR for review purposes (in future please could you create the PR and upload the link to Aptem)

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -6,7 +6,27 @@
...
 def index():
-    return 'Hello World!'
+    items = sorted(session.get_items(), key=lambda i: i['status'], reverse=True)
+    return render_template('index.html', items=items)
+    
+@app.route('/', methods=['POST'])
+def add_item():
+    title = request.form['item_title']
+    session.add_item(title)
+    return redirect(url_for('index'))
```
Comment (Jonesey13) [inline app.py:16]: Good job for spotting you should use a redirect!

File: session_items.py [hunk 2]
```diff
@@ -67,3 +67,16 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+def remove_item(item):
+    """
+    Removes an existing item in the session. If no existing item matches the ID of the specified item, nothing is removed.
```
Comment (Jonesey13) [inline session_items.py]: Technically this checks if the item's attributes match (so just having the same id does not suffice)

File: session_items.py [hunk 3]
```diff
@@ -67,3 +67,16 @@ def save_item(item):
...
+
+    Args:
+        item: The item to remove.
+    """
+    existing_items = get_items()
```
Comment (Jonesey13) [inline session_items.py]: How would you achieve the same result using a list comprehension?

File: templates/index.html [hunk 4]
```diff
@@ -10,8 +10,30 @@ <h1 class="display-4">To-Do App</h1>
   <div class="row">
     <div class="col-md-12">
       <h2>Items</h2>
+  
+      <form action="{{ url_for('index') }}" method="post" class="input-group mb-3">
+        <input type="text" id="title" name="item_title" class="form-control" />
```
Comment (Jonesey13) [inline templates/index.html]: Probably want to add a "required" attribute here

File: templates/index.html [hunk 5]
```diff
@@ -10,8 +10,30 @@ <h1 class="display-4">To-Do App</h1>
...
-        <!-- List the todo items here -->
+        {% for item in items %}
+          <li class="list-group-item">
+            <b>{{ item.title }}</b>: {{ item.status }}
+            <div class="float-right">
+              <form action="{{ url_for('delete_item', id=item.id) }}" method="post" class="d-inline">
+                <button type="submit" class="btn btn-danger mr-1">Delete</button>
+              </form>
+              {% if item.status == 'Not Started' %}
+              <form action="{{ url_for('mark_item_as_complete', id=item.id) }}" method="post" class="d-inline">
```
Comment (Jonesey13) [inline templates/index.html]: Indentation

File: session_items.py (hunk 3 above)
Comment (chengjoanna1729) [reply session_items.py]: as in something like `session['items'] = [listItem for listItem in getItems() if listItem != item]` as the whole method?

**Comment by Jonesey13** [comment]
This all looks functionally sound to me! 

There's a few style comments but this is ready for merging in. :)

File: session_items.py (hunk 3 above)
Comment (Jonesey13) [reply session_items.py]: Yes :)

**Pull request merged**